File: src/main/java/com/xyfindables/ui/views/XYAwesomeTextView.java
package com.xyfindables.ui.views;


import android.content.Context;
import android.support.annotation.NonNull;
import android.util.AttributeSet;
import android.view.ContextThemeWrapper;

import com.xyfindables.ui.R;
import com.xyfindables.ui.XYGlobalFonts;

/**
 * Created by arietrouw on 1/15/17.
 */

public class XYAwesomeTextView extends XYTextView {
    public XYAwesomeTextView(Context context) {
        this(context, null, 0);
    }

    public XYAwesomeTextView(Context context, @NonNull AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public XYAwesomeTextView(Context context, @NonNull AttributeSet attrs, int defStyle) {
        super(new ContextThemeWrapper(context, R.style.xy_awesome_textview), attrs, defStyle);
        XYGlobalFonts.setViewFontAwesome(context, this);
    }
}

File: ui-android-library/src/main/java/com/xyfindables/ui/views/XYAwesomeTextView.kt
package com.xyfindables.ui.views


import android.content.Context
import android.util.AttributeSet
import android.view.ContextThemeWrapper

import com.xyfindables.ui.R
import com.xyfindables.ui.XYGlobalFonts

/**
 * Created by arietrouw on 1/15/17.
 */

class XYAwesomeTextView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : XYTextView(ContextThemeWrapper(context, R.style.xy_awesome_textview), attrs, defStyle) {

    init {
        XYGlobalFonts.setViewFontAwesome(context, this)
    }
}
